### ISSUE TITLE
About the computation of the file `norm_params.json`

### ISSUE THREAD

**Issue by whpy** [tung-nd/IndiBench#3]
Thank you for your works!

I am trying to reproduce the demo in your paper, and I have a question about how the mean and standard deviation in norm_param.json are computed. 

I read the paper carefully but couldn’t find an explicit description of how these statistics are calculated. My current guess is that the mean and standard deviation are computed across both spatial and temporal dimensions over the dataset (which is also mentioned in Appendix 10.1 about how to calculate the mean of values for climatology). However, when I try to reproduce them in this way, I find that the mean values for the variables UGRD, VGRD, VGRD_prl500, VGRD_prl850, and VGRD_prl925 differ a lot from those in `norm_param.json`, while the other variables match reasonably well, where the relative errors are all around 10%.

Could you please clarify:

- 1 Over which dimensions you compute the mean and standard deviation (e.g., space, time, pressure levels, ensemble members, etc.)?

- 2 Which subset of data you use (e.g., only training years, specific times, any masking or filtering)?

Thank you very much for your help!

**Comment by whpy** [comment]
Here is my code for reproduction:
```python
import india_benchmark
import h5py
import json
import numpy as np
import copy

training_dataset_path = 'path/to/IndianWeatherBench/train'

bch_path = 'path/to/norm_params.json'
with open(bch_path) as f:
    bch_mark = json.load(f)['mean']

variables_results = copy.deepcopy(bch_mark)
for k in variables_results:
    variables_results[k] = 0
var_name_list = list(bch_mark.keys())

# they are different
# print('variable results',variables_results)
# print('benchmark',bch_mark)

years = [i for i in range(2000, 2018)]
months = [i for i in range(1,13)]
days = [i for i in range(1,32)]
hours = [0, 1, 2, 3]


variables = ['TMP']

legal_days1 = [-1,31,29,31,30,31,30,31,31,30,31,30,31]
legal_days2 = [-1,31,28,31,30,31,30,31,31,30,31,30,31]

# calculate the mean value
count = 0
mean = 0
for y in years:
    for m in months:
        for d in days:
            for h in hours:
                if y%4 == 0:
                    legal_day = legal_days1[m]
                else:
                    legal_day = legal_days2[m]
            if d > legal_day:
                print(f'skip {y}-{m}-{d}')
                continue
            else:
                file_path = f'{training_dataset_path}/{y}-{m:0>2d}-{d:0>2d}_{h:0>2d}.h5'
            file = h5py.File(file_path)
            for var in var_name_list:
                v = file[var][()]
                variables_results[var] += v.mean()
            count += 1

print(variables_results['VGRD_prl600']/count)
print(bch_mark['VGRD_prl600'])
for var in var_name_list:
    if 'VGRD' in var or 'UGRD' in var:
        print(f'The relative error of mean value for variable {var} is: {np.abs( (variables_results[var]/count - bch_mark[var]) /bch_mark[var] )*100:.2f}% (ours: {variables_results[var]/count} bch: {bch_mark[var]})')
        print(f'{file[var][()].mean()}')
    else:
        print(f'The relative error of mean value for variable {var} is: {np.abs( (variables_results[var]/count - bch_mark[var]) /bch_mark[var] )*100:.2f}%')

variables_mean_values = copy.deepcopy(variables_results)
for k in variables_mean_values:
    variables_mean_values[k] = variables_mean_values[k]/count

# calculate the standard deviation
with open(bch_path) as f:
    bch_mark_std = json.load(f)['std']

variables_std_values = copy.deepcopy(bch_mark_std)
for k in variables_std_values:
    variables_std_values[k] = 0

count = 0
for y in years:
    for m in months:
        for d in days:
            for h in hours:
                if y%4 == 0:
                    legal_day = legal_days1[m]
                else:
                    legal_day = legal_days2[m]
            if d > legal_day:
                print(f'skip {y}-{m}-{d}')
                continue
            else:
                file_path = f'{training_dataset_path}/{y}-{m:0>2d}-{d:0>2d}_{h:0>2d}.h5'
            file = h5py.File(file_path)
            for var in var_name_list:
                v = file[var][()]
                variables_std_values[var] += np.square(v-variables_mean_values[var]).mean()
                # print(np.square(v-variables_mean_values[var]).mean())
            count += 1

for k in variables_std_values:
    variables_std_values[k] = np.sqrt(variables_std_values[k]/count)
for var in var_name_list:
    print(f'The relative error of standard deviation for variable {var} is: {np.abs( ( variables_std_values[var] - bch_mark_std[var]) /bch_mark_std[var] )*100:.2f}% (ours: {variables_std_values[var]} bch: {bch_mark_std[var]})')

```
we can see that most of the standard deviations could match well with relative errors less than 5%.

**Comment by tung-nd** [comment]
Have you been able to resolve this?
I computed the statistics across the spatial and temporal dimensions of the training data. For pressure-level variables, the statistics were computed for each level individually.
